core/app/models/spree/stock/simple_coordinator.rb

```diff
@@ -52,7 +52,9 @@ def build_shipments
         on_hand_packages, backordered_packages, leftover = @allocator.allocate_inventory(@desired)
 
         unless leftover.empty?
-          raise Spree::Order::InsufficientStock
+          item_names = leftover.variants.map(&:name).to_sentence
+          message = I18n.t('spree.inventory_error_flash_for_insufficient_quantity', names: item_names)
+          raise Spree::Order::InsufficientStock.new(message)
```

Comment:
I like the idea but I think we can take advantage of creating a custom exception that contains the unavailable items and let the consumer decide what to do with those. 

What I have in mind is changing the [exception definition](https://github.com/solidusio/solidus/blob/78ccf7f7ca6ac625328a9211225a17d71ab3c991/core/app/models/spree/order.rb#L29) to something like:

```ruby
class InsufficientStock < StandardError
  attr_reader :items

  def initialize(message=nil, items: [])
    super(message)

    @items = items
  end
end
```

and push names in it with:

```ruby
raise Spree::Order::InsufficientStock.new(items: leftovers)
```

At that point, we can consume the exception data in different ways depending on the context, even using `rescue_from` since its [handler methods accept an argument](https://apidock.com/rails/ActiveSupport/Rescuable/ClassMethods/rescue_from) that will be the exception containing the items. 

```ruby
rescue_from Spree::Order::InsufficientStock, with: :something 

def something(exception)
  # exception.items is available here!
end
```

What do you think?

Reply:
I like it! will refactor my code to take this path